UnityProject/Assets/Scripts/NPC/MobSprite.cs

```diff
@@ -0,0 +1,177 @@
+﻿using UnityEngine;
+using Mirror;
+using Core.Directionals;
+using Effects.Overlays;
+
+namespace Systems.Mob
+{
+	/// <summary>
+	/// Easy to use Directional sprite handler for mobs
+	/// </summary>
+	[RequireComponent(typeof(DirectionalSpriteV2))]
+	public class MobSprite : NetworkBehaviour
+	{
+		[Header("References")]
+		[SerializeField]
+		private SpriteHandler spriteHandler = default;
+		[SerializeField]
+		private SpriteRenderer spriteRend = default;
+
+		[Header("Settings")]
+		[SerializeField]
+		private int aliveSpriteIndex = 0;
+		[SerializeField]
+		private int deadSpriteIndex = 1;
+
+		[Tooltip("Assign the prefab responsible for the burning overlay.")]
+		[SerializeField]
+		private GameObject burningPrefab = default;
+
+		[Tooltip("Assign the prefab responsible for the electrocuted overlay.")]
+		[SerializeField]
+		private GameObject electrocutedPrefab = default;
+
+		private LivingHealthBehaviour health;
+		private Directional directional;
+		private DirectionalSpriteV2 directionalSprite;
+
+		private PlayerDirectionalOverlay burningOverlay;
+		private PlayerDirectionalOverlay electrocutedOverlay;
+
+		[SyncVar(hook = nameof(SyncRotChange))] private float spriteRot;
+
+		#region Lifecycle
+
+		private void OnEnable()
+		{
+			EnsureInit();
+		}
+
+		private void EnsureInit()
+		{
+			if (health != null) return;
+			health = GetComponent<LivingHealthBehaviour>();
+			directional = GetComponent<Directional>();
+			directionalSprite = GetComponent<DirectionalSpriteV2>();
+
+			directional.OnDirectionChange.AddListener(OnDirectionChange);
+		}
+
+		public override void OnStartServer()
+		{
+			EnsureInit();
+			spriteRot = 0;
+		}
+
+		public override void OnStartClient()
+		{
+			AddOverlayGameObjects();
+			EnsureInit();
+			SyncRotChange(spriteRot, spriteRot);
+		}
+
+		/// <summary>
+		/// Instantiate and attach the sprite overlays if they don't exist
+		/// </summary>
+		private void AddOverlayGameObjects()
+		{
+			if (burningOverlay == null)
+			{
+				burningOverlay = Instantiate(burningPrefab, transform).GetComponent<PlayerDirectionalOverlay>();
+				burningOverlay.enabled = true;
+				burningOverlay.StopOverlay();
+			}
+			if (electrocutedOverlay == null)
+			{
+				electrocutedOverlay = Instantiate(electrocutedPrefab, transform).GetComponent<PlayerDirectionalOverlay>();
+				electrocutedOverlay.enabled = true;
+				electrocutedOverlay.StopOverlay();
+			}
+		}
+
+		#endregion Lifecycle
+
+		//The local rotation of the sprite obj
+		private void SyncRotChange(float oldRot, float newRot)
```

Comment:
Codacy found an issue: [Remove this unused method parameter 'oldRot'.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6356172)

Reply:
`SyncVar`.